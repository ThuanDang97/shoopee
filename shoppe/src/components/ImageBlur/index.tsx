import { memo } from 'react'
import Image, { ImageLoader, ImageProps } from 'next/image'

// Utils
import { shimmer, toBase64 } from '@utils/index'

// Services
import imageLoader from '@services/imageLoader'

const ImageBlur = ({ width, height, ...rest }: ImageProps) => {
  return (
    <Image
      width={width}
      height={height}
      placeholder="blur"
      loader={imageLoader as ImageLoader}
      blurDataURL={`data:image/svg+xml;base64,${toBase64(
        shimmer(width as number, height as number),
      )}`}
      unoptimized
      {...rest}
    />
  )
}

export default memo(ImageBlur)
