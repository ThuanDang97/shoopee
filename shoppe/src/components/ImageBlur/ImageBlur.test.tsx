import React from 'react'

// Component
import ImageBlur from '@components/ImageBlur'

// Utils
import { render } from '@utils/testUtils'

describe('Image render', () => {
  const props = {
    src: '/facebook.webp',
    alt: 'facebook',
    width: 200,
    height: 200,
  }
  it('Should show match Image DOM Snapshot', () => {
    const { container } = render(<ImageBlur {...props} />)
    expect(container).toMatchSnapshot()
  })
})
