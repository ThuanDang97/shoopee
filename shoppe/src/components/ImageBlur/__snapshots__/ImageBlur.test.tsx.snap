// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image render Should show match Image DOM Snapshot 1`] = `
<div>
  <img
    alt="facebook"
    data-nimg="1"
    decoding="async"
    height="200"
    loading="lazy"
    src="/facebook.webp"
    style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
    width="200"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
